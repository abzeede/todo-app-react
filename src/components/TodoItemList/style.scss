.todo-item-list {
  padding-inline-start: 0;
  margin-block-start: 0;
  margin-block-end: 0;
}
