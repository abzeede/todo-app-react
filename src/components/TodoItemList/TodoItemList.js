import React from 'react'
import { shape, arrayOf, string, bool, number, func } from 'prop-types'
import TodoItem from 'components/TodoItem'
import './style.scss'

const TodoItemList = ({ items, ...props }) => {
  return (
    <ol className="todo-item-list">
      {items.map(item => (
        <TodoItem key={item.id} {...props} {...item} />
      ))}
    </ol>
  )
}

TodoItemList.propTypes = {
  items: arrayOf(
    shape({
      id: number,
      detail: string,
      done: bool,
    }),
  ).isRequired,
  onDelete: func.isRequired,
  onToggleDone: func.isRequired,
  onEdit: func,
}

TodoItemList.defaultProps = {
  items: [],
  onEdit: () => {},
}

export default TodoItemList
