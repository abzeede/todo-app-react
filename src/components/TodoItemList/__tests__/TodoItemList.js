import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import TodoItemList from '../TodoItemList'

describe('TodoItemList', () => {
  it('should render properly', () => {
    const items = [
      { id: 1, detail: 'todo 1', done: false },
      { id: 2, detail: 'todo 2', done: false },
      { id: 3, detail: 'todo 3', done: false },
    ]
    const wrapper = mount(<TodoItemList items={items} onDelete={() => {}} onToggleDone={() => {}} />)
    expect(wrapper.find('.todo-item')).toHaveLength(3)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
