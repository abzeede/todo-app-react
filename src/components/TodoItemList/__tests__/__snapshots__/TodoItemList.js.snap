// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItemList should render properly 1`] = `
<TodoItemList
  items={
    Array [
      Object {
        "detail": "todo 1",
        "done": false,
        "id": 1,
      },
      Object {
        "detail": "todo 2",
        "done": false,
        "id": 2,
      },
      Object {
        "detail": "todo 3",
        "done": false,
        "id": 3,
      },
    ]
  }
  onDelete={[Function]}
  onEdit={[Function]}
  onToggleDone={[Function]}
>
  <ol
    className="todo-item-list"
  >
    <TodoItem
      detail="todo 1"
      done={false}
      id={1}
      key="1"
      onDelete={[Function]}
      onEdit={[Function]}
      onToggleDone={[Function]}
    >
      <li
        className="todo-item"
      >
        <span
          className="todo-item__checkbox"
        >
          <Checkbox
            checked={false}
            onClick={[Function]}
          >
            <span
              className="checkbox false"
              onClick={[Function]}
            />
          </Checkbox>
        </span>
        <input
          className="todo-item__input "
          onChange={[Function]}
          value="todo 1"
        />
        <button
          className="todo-item__remove-btn"
          onClick={[Function]}
        >
          <i
            className="fa fa-trash"
          />
        </button>
      </li>
    </TodoItem>
    <TodoItem
      detail="todo 2"
      done={false}
      id={2}
      key="2"
      onDelete={[Function]}
      onEdit={[Function]}
      onToggleDone={[Function]}
    >
      <li
        className="todo-item"
      >
        <span
          className="todo-item__checkbox"
        >
          <Checkbox
            checked={false}
            onClick={[Function]}
          >
            <span
              className="checkbox false"
              onClick={[Function]}
            />
          </Checkbox>
        </span>
        <input
          className="todo-item__input "
          onChange={[Function]}
          value="todo 2"
        />
        <button
          className="todo-item__remove-btn"
          onClick={[Function]}
        >
          <i
            className="fa fa-trash"
          />
        </button>
      </li>
    </TodoItem>
    <TodoItem
      detail="todo 3"
      done={false}
      id={3}
      key="3"
      onDelete={[Function]}
      onEdit={[Function]}
      onToggleDone={[Function]}
    >
      <li
        className="todo-item"
      >
        <span
          className="todo-item__checkbox"
        >
          <Checkbox
            checked={false}
            onClick={[Function]}
          >
            <span
              className="checkbox false"
              onClick={[Function]}
            />
          </Checkbox>
        </span>
        <input
          className="todo-item__input "
          onChange={[Function]}
          value="todo 3"
        />
        <button
          className="todo-item__remove-btn"
          onClick={[Function]}
        >
          <i
            className="fa fa-trash"
          />
        </button>
      </li>
    </TodoItem>
  </ol>
</TodoItemList>
`;
